ipa-core/src/protocol/ipa_prf/prf_sharding/mod.rs

```diff
@@ -1068,4 +1071,54 @@ pub mod tests {
             assert_eq!(result, &expected);
         });
     }
+
+    #[test]
+    fn capping_bugfix() {
+        run(|| async move {
+            let world = TestWorld::default();
+
+            let records: Vec<PreShardedAndSortedOPRFTestInput<BA8, BA3, BA20>> = vec![
+                /* First User */
+                oprf_test_input_with_timestamp(10251308645, false, 218, 0, 29135),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 3, 179334),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 3, 313530),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 5, 327956),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 6, 368968),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 1, 453715),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 2, 471555),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 6, 530458),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 6, 536295),
+                oprf_test_input_with_timestamp(10251308645, true, 0, 6, 600360),
+            ];
+
+            let mut expected = [0_u128; 256];
+            expected[218] = 32;
+
+            let histogram = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
```

Comment:
could be a const

like `const HISTOGRAM: usize = [1; 10];`

But I guess that you really want `let histogram = [1; records.len()];`